src/Blaster.WebApi/Features/CapabilityDashboard/main.js

```diff
@@ -20,7 +20,52 @@ const app = new Vue({
     },
     computed: {
         capabilityFound: function() {
-            return this.capability != null && this.capability.id != null 
+            return this.capability != null && this.capability.id != null
+        },
+        isLegacyComputed: function () { // Determine if Capability is v1 or v2
+            if (this.capability) // Ensure capability object exists
+            {
+                if (this.capability.rootId) { // Check if rootId is null
+                    if (this.capability.rootId === "") {
+                        return true;
+                    } else {
+                        return false;
+                    }
+                }
+                return true;
+            }
+        },
+        isJoinedComputed: function () {
+            var isMemberRawText = this.getMembershipStatusFor();
+            if (isMemberRawText === "member") {
+                return true;
+            } else {
+                return false;
+            }
+        },
+        isAddContextDisallowedComputed: function() {
+            var isLegacy = this.isLegacyComputed;
+            var isJoined = this.isJoinedComputed;
+
+            if (isLegacy == false && isJoined == true) {
+                return false;
+            } else {
+                return true;
+            }
+        },
+        disabledContextButtonReasonComputed: function() {
+            if (this.isAddContextDisallowedComputed) {
+                var msg = "Reason(s) why this button is disabled:" + "\n";

```

Comment:
Bit convoluted language, but no big deal. I wonder if it makes sense to allow people to join a legacy capability but we can look at that later...

no changes needed here

Reply:
All right!